x-pack/plugins/reporting/server/routes/lib/authorized_user_pre_routing.test.ts

```diff
@@ -82,82 +93,101 @@ describe('authorized_user_pre_routing', function () {
       const mockHandler = () => {
         throw new Error('Handler callback should not be called');
       };
+
       expect(
-        authorizedUserPreRouting(mockCore, mockHandler)(
+        await authorizedUserPreRouting(mockCore, mockHandler)(
           getMockContext(),
           getMockRequest(),
-          mockResponseFactory
+          getMockResponseFactory()
         )
       ).toMatchObject({ body: `Sorry, you don't have access to Reporting` });
     });
 
-    it('should return from handler when security is enabled and user has explicitly allowed role', function (done) {
-      mockCore.getPluginSetupDeps = () =>
-        ({
-          // @ts-ignore
-          ...mockCore.pluginSetupDeps,
+    it('should return from handler when security is enabled and user has explicitly allowed role', async function () {
+      mockStartDeps = await createMockPluginStart(
+        {
           security: {
-            license: { isEnabled: () => true },
             authc: {
               getCurrentUser: () => ({ username: 'friendlyuser', roles: ['reporting_user'] }),
             },
           },
-        } as unknown as ReportingInternalSetup);
-      const mockResponseFactory = getMockResponseFactory();
+        },
+        mockReportingConfig
+      );
+      mockCore = await createMockReportingCore(mockReportingConfig, mockSetupDeps, mockStartDeps);
 
-      authorizedUserPreRouting(mockCore, (user) => {
+      let handlerCalled = false;
+      await authorizedUserPreRouting(mockCore, (user: unknown) => {
         expect(user).toMatchObject({ roles: ['reporting_user'], username: 'friendlyuser' });
-        done();
+        handlerCalled = true;
         return Promise.resolve({ status: 200, options: {} });
-      })(getMockContext(), getMockRequest(), mockResponseFactory);
+      })(getMockContext(), getMockRequest(), getMockResponseFactory());
+      expect(handlerCalled).toBe(true);
     });
 
-    it('should return from handler when security is enabled and user has superuser role', async function () {});
+    it('should return from handler when security is enabled and user has superuser role', async function () {
+      mockStartDeps = await createMockPluginStart(
+        {
+          security: {
+            authc: { getCurrentUser: () => ({ username: 'friendlyuser', roles: ['superuser'] }) },
+          },
+        },
+        mockReportingConfig
+      );
+      mockCore = await createMockReportingCore(mockReportingConfig, mockSetupDeps, mockStartDeps);
+
+      let handlerCalled = false;
+      await authorizedUserPreRouting(mockCore, (user: unknown) => {
+        expect(user).toMatchObject({ roles: ['superuser'], username: 'friendlyuser' });
+        handlerCalled = true;
+        return Promise.resolve({ status: 200, options: {} });
+      })(getMockContext(), getMockRequest(), getMockResponseFactory());
+      expect(handlerCalled).toBe(true);
```

Comment:
It is a good practice to write declarative tests so we can easily catch and debug errors in the code.

```suggestion
      const handler = jest.fn().mockResolvedValue({ status: 200, options: {} });
      await authorizedUserPreRouting(mockCore, handler)(
        getMockContext(),
        getMockRequest(),
        getMockResponseFactory()
      );
      
      expect(handler).toHaveBeenCalled();

      const [[user]] = handler.mock.calls;
      expect(user).toMatchObject({ roles: ['superuser'], username: 'friendlyuser' });
```